addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -1250,28 +1250,49 @@ export class Rtc extends Record {
     }
 
     logSnapshot() {
-        const peers = [];
-        this.p2pService?.peers.forEach((peer) => {
-            peers.push({
-                id: peer.id,
-                state: peer.connection.connectionState,
-                iceState: peer.connection.iceConnectionState,
-            });
-        });
         const server = {};
         if (this.state.connectionType === CONNECTION_TYPES.SERVER) {
             server.info = this.serverInfo;
             server.state = this.sfuClient?.state;
-            server.consumersStats = this.state.channel.rtcSessions.map((session) =>
-                this.network.getSfuConsumerStats(session.id)
-            );
             server.errors = this.sfuClient?.errors.map((error) => error.message);
         }
+        const sessions = this.state.channel?.rtcSessions.map((session) => {
+            const sessionInfo = {
+                id: session.id,
+                channelMemberId: session.channel_member_id?.id,
+                state: session.connectionState,
+                audioError: session.audioError,
+                videoError: session.videoError,
+                sfuConsumers: this.network.getSfuConsumerStats(session.id),
+            };
+            if (session.eq(this.localSession)) {
+                sessionInfo.isSelf = true;
+            }
```

Comment:
Shouldn't this be `selfSession` instead? (so the data is the same when retrieving the session data of self user, regardless of tab doing the call).

Reply:
@alexkuhn logs must come from the call tab, otherwise most of the data will be empty, maybe later we could make it so that before downloading we merge the logs from each tabs

Reply:
I would suggested to still change to `selfSession` there, even though it's the same it prepares a bit for handling download of logs in any tab.

If the logs are intended to be download from the call tab, I don't see it being restricted in the call settings template. We should limit it (and maybe put some guidance when attempting to download logs from a tab not making the call?).

Reply:
@alexkuhn :+1:  updated,

I also restricted the snapshot to be made only if we are in a call, as there is nothing to collect if not in a call (as the state of rtc_service is cleaned after a call),

To give more context, a snapshot is made right before the end of calls and when the download button is clicked, there can be different logs in different tabs if multiple calls have been made in different tabs (or transferred), it should be relatively easy to collect all logs from all tabs and merge them (as there shouldn't be key collision) before downloading.

Reply:
Having call logs spread in different tabs and needing to manually download each of them and reconcile by hand is fastidious labour. It's important to ease retrievable of logs in any tabs as much as we can.

What about sending the store logs to shared worker and/or saving in local storage?

Reply:
@alexkuhn in most cases, just downloading 1 log (from the tab where the problem happened) is enough, but I will make a poc for a unified log download, I was thinking of collecting just-in-time (with the cross rtc cross tab system) instead of using shared memory.

Reply:
Ok, let's merge this one as this is better logs. Thanks again :)